std/assembly/typedarray.ts

```diff
@@ -1364,6 +1408,38 @@ function MAP<TArray extends ArrayBufferView, T>(
   return out;
 }
 
+// @ts-ignore: decorator
+@inline
+function FILTER<TArray extends ArrayBufferView, T>(
+  array: TArray,
+  fn: (value: T, index: i32, self: TArray) => bool,
+): TArray {
+  var len = array.length;
+  var out = instantiate<TArray>(len);
```

Comment:
would be cleaner to allocate a buffer, fill it, realloc it, and then build a `TArray` from it directly. minimal improvement, but saves reassigning it after realloc.

Reply:
Yes. Also it skip unnecessary `memory.fill`